app/services/check_course_jobs.rb

```diff
@@ -30,21 +48,43 @@ def delete_unique_lock
   def expected_digest
     hash = {
       'class' => 'CourseDataUpdateWorker',
-      'queue' => queue_for(@course),
+      'queue' => @queue,
       'unique_args' => [@course_id]
     }.to_json
     digest = OpenSSL::Digest::MD5.hexdigest hash
     "uniquejobs:#{digest}"
   end
 
-  def find_job
+  def find_job # rubocop:disable Metrics/CyclomaticComplexity
+    Sidekiq::ScheduledSet.new.select do |retri|
```

Comment:
What is this doing?

Reply:
I am basically searching for scheduled jobs here, though it doesn't seem like the use of `CourseDataUpdateWorker` is scheduled. I am still trying to understand what is going on regarding the issue I mentioned on slack, the use of this is still somewhat unclear to me. [https://github.com/mperham/sidekiq/wiki/API#named-queues](https://github.com/mperham/sidekiq/wiki/API#named-queues)

Reply:
I think this is exclusively for "Scheduled Jobs", which are jobs that sidekiq will run at a particular time. See https://github.com/mperham/sidekiq/wiki/Scheduled-Jobs

We don't use time-based scheduling for anything at this point, so I think you can remove this.

Reply:
Yes, removed